src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectSnapshotManager.cs

```diff
@@ -17,6 +17,8 @@ internal abstract class ProjectSnapshotManager : ILanguageService
 
         public abstract ProjectSnapshot GetLoadedProject(string filePath);
 
+        public abstract ProjectSnapshot GetLoadedProject(ProjectId projectId);
+
         public abstract ProjectSnapshot GetOrCreateProject(string filePath);
```

Comment:
Can this take ProjectId?

Reply:
Sadly not because in the case that a project doesn't exist we need to create an ephemeral project with the provided file path.